plugins/UsersManager/API.php

```diff
@@ -780,7 +819,23 @@ private function isUserTheOnlyUserHavingSuperUserAccess($userLogin)
     }
 
     /**
-     * Generates a unique MD5 for the given login & password
+     * Generates a new random authentication token.
+     *
+     * @param string $userLogin Login
+     * @return string
+     */
+    public function createTokenAuth($userLogin)
+    {
+        return hash(
```

Comment:
I'm now wondering whether we should maybe still go with `md5` and never regenerate token when `UsersManager.updateUser` is being called. There is not really a benefit of `sha256` over `md5` I'd say as long as the generation is based on random input. 

The `sha256` as discussed in the issue makes more sense when we store the md5 token hashed as sha256. This way the token will be no longer stored as plain text which is a big problem right now. This is a feature for another PR though and will not be in the 3.0 release. Still going with `md5` could allow us later to differentiate between a plain md5 token with 32 char length and a sha256 hashed md5 token (which is 64 length hashed in db). We will likely store auth tokens hashed as part of  https://github.com/piwik/piwik/issues/6559 . When implementing this we will likely use a new table for storing auth tokens (app specific tokens/passwords) anyway and we can also hash all tokens on "update" and write them into a different table eg `user_tokens`. So this differentation between md5 and sha256 will be likely not needed actually and we can keep using sha256 in this PR for token generation 👍  So I presume no change needed, just wanted to write down what came to my mind before :)